pkgs/by-name/yk/ykush/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  libusb1,
+  runCommand,
+  gcc,
+}:
+stdenv.mkDerivation (final: {
+  pname = "ykushcmd";
+  version = "1.3.0";
+
+  buildInputs = [ libusb1 ];
+  nativeBuildInputs = [ gcc ];
+
+  src = fetchFromGitHub {
+    owner = "yepkit";
+    repo = "ykush";
+    rev = "v${final.version}";
+    hash = "sha256-ykZAuHk3zPiH20yMYPgq6kJv1LAhUko6jJfAqGREvWA=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D bin/ykushcmd $out/bin/ykushcmd
+    install -Dm444 ${./99-ykush.rules} $out/lib/udev/rules.d/99_ykush.rules
+
+    runHook postInstall
+  '';
+
+  passthru.tests = {
+    # We can't run any actual tests without hardware, but we can at least check the binary.
+    run-only = runCommand "${final.pname}-test" ''
+      ${final}/bin/ykushcmd -h | grep YKUSHCMD
```

Comment:
This grep seems mildly fragile. Is there a failure mode where the binary executes, exits with an exit code of 0 and does not work as intended?

Reply:
For this version, this is matching against known help text -- if executable, the binary should always output this when provided with `-h`:

https://github.com/Yepkit/ykush/blob/8268c917bb054efb3fa7eaf961e104ab8df8d577/src/help/ykush_help.cpp#L30

Upstream can break this if they change their version string format; but upstream can pretty much always break test behaviors, and it seems better than no test at all. (Every option other than this one requires hardware; and there's no discrete --version command. T_T)

Reply:
Oh wow i just checks, it exits with an error code if you call it with -h. So, uh, grep is probably indeed required.